doc/headset.md

```diff
@@ -0,0 +1,430 @@
+# Genesys Cloud WebRTC SDK Headset Utility
+
+## SDK Headset Index
+* [Purpose of SdkHeadset]
+* [SdkHeadset]
+    * [Example usage]
+    * [Methods]
+    * [Events]
+
+
+## Purpose of SdkHeadset
+The SdkHeadset namespace connects directly to the new headset library that allows device call controls from at this moment three headset vendors. Plantronics/Poly, Sennheiser/EPOS and Jabra. It makes it possible for events from the headset to be properly handled and reflected in consuming apps. It also allows events from consuming apps to be properly reflected in the headset itself. This ensures the states of both the devices and consuming apps are always in sync
+
+--------
+
+## SdkHeadset
+
+### Example Usage
+
+``` ts
+
+/* async function for demonstration */
+async function testAnswerCall () {
+    /* This function handles a fromHeadset event. This will not cause the headset library to be called */
+    sdk.acceptPendingSession('123456789', 'softphone', true);
+    /*  async acceptPendingSession (params: IPendingSessionActionParams):       Promise<void> {
+        !params.fromHeadset && this.headset.answerIncomingCall(params.      conversationId);
+        await this.sessionManager.proceedWithSession(params);
+    }
+  */
+}
+
+async function testMuteCall () {
+    /* This function handles a from app event. This will cause the headset library to be called */
+    sdk.setAudioMute ('123456789', 'softphone', false | undefined);
```

Comment:
I'm almost certain all the headset events have the potential to call through the headset class. The only one that may not be bidirectional (headset to app and vice versa) is reject call because I'm honestly not too sure how one would do that from the headset. But for these examples, I just wanted to show the results of having `fromHeadset` being different. So in these cases `fromHeadset` for `setAudioMute` is false so it will call the headset library to properly update the headset's state.  While `acceptPendingSession` has a `fromHeadset` of true, meaning the headset already knows about the state change so we wouldn't need to call it again.  I can remove those comments though because I definitely see how it can muddy things up.